docs/flow/config-language/components.md

```diff
@@ -6,3 +6,96 @@ weight: 300
+
+Under the hood, components are orchestrated via the [_component
+controller_]({{< relref "../concepts/component_controller.md" >}}), who is
+responsible for scheduling them, reporting their health and debug status, and
+continuously re-evaluating their inputs and outputs.
+
+
+## Configuring components
+Each top-level River _block_ will instantiate a new component. All components
+are identified by their name, describing what the component is responsible for,
+while some allow or require to provide an extra user-specified _label_.
+
+You can see a list of all available components [here]({{< relref "../components/_index.md" >}}).
+Each one features a complete reference page, so getting a component to work for
+you should be as easy as reading its documentation and copy/pasting from an
+example.
+
+## Arguments and Exports
+Most user interactions with components will center around two basic concepts;
+_Arguments_ and _Exports_.
+
+* _Arguments_ are settings which modify the behavior of a component. They can
+ be any number of attributes or nested unlabeled blocks, some of them being
+required and some being optional. Any optional arguments that are not set, will
+take on their default values.
+
+* _Exports_ are zero or more output values that can be referred to by other
+  components, and can be of any River type.
+
+Here's a quick example; the following block defines a `local.file` component
+labeled "targets". The `local.file.targets` component will then expose the
+file `content` as a string in its Exports.
+
+The `filename` attribute is a _required_ argument; the user can also define a
+number of _optional_ ones, in this case `detector`, `poll_frequency` and
+`is_secret`, which configure how and how often the file should be polled
+as well as whether its contents are sensitive or not.
+
+```river
+local.file "targets" {
+	// Required Argument
+	filename = "/etc/agent/targets" 
+
+	// Optional Arguments
+	//   is_secret      = <boolean>
+	//   detector       = < "fsnotify" | "poll" >
+	//   poll_frequency = <duration> 
+
+	// Exports: a single field named `content`
+	// It can be referred to as `local.file.token.content`
+}
+```
+
+## Referencing components
+To wire components together, one can use the Exports of one as the Arguments
+to another.
+
+For example, here's a component that scrapes Prometheus metrics. The `targets`
+field is populated with two scrape targets; a constant one `localhost:9001` and
+an expression that ties the target to the value of
+`local.file.target.content`.
+
+```river
+prometheus.scrape "default" {
+	targets = [
+		{ "__address__" = local.file.target.content }, // tada!
+		{ "__address__" = "localhost:9001" },
+	] 
+
+	forward_to = [prometheus.remote_write.default.receiver]
+	scrape_config {
+		job_name = "default"
+	}
+}
+```
+
+Every time the file contents change and the `local.file` component re-evaluates
+its Exports, the value will propagated to `prometheus.scrape` so it can start
+scraping the new target.
```

Comment:
nit: New exports get exported/exposed/provided, but not reevaluted: 

> Every time the file changes on disk, the `local.file` component will re-read the file and expose the latest contents through its exports. This event will be propagated to the `prometheus.scrape` referencing the file so it can start scraping the new target.